strings/strip_url_params.py

```diff
@@ -0,0 +1,106 @@
+"""
+Write a function that does the following:
+Removes any duplicate query string parameters from the url
+Removes any query string parameters specified within the 2nd argument (optional array)
+
+An example:
+www.saadbenn.com?a=1&b=2&a=2') // returns 'www.saadbenn.com?a=1&b=2'
+"""
+
+# Here is a very non-pythonic grotesque solution
+from collections import defaultdict
+def strip_url_params(url, params_to_strip=[]):
```

Comment:
please use `None` as default value for arguments in any function, rather than a mutable type (like `[]`). 
and since there are 3 algorithms, use names like `strip_url_params1`, `strip_url_params2`, etc.